File: src/main/java/org/checkerframework/checker/mungo/typestate/graph/exceptions/UnusedStates.java
package org.checkerframework.checker.mungo.typestate.graph.exceptions;

import org.checkerframework.checker.mungo.typestate.ast.TStateNode;

import java.util.List;
import java.util.stream.Collectors;

public class UnusedStates extends TypestateError {

  public static final long serialVersionUID = 0L;

  private final List<TStateNode> unusedStates;

  public UnusedStates(List<TStateNode> unusedStates) {
    this.unusedStates = unusedStates;
  }

  @Override
  public String getMessage() {
    return String.format("Unused states in %s: %s", unusedStates.get(0).pos.getBasename(), unusedStates.stream().map(s -> s.name).collect(Collectors.joining("; ")));
  }

}

File: src/main/kotlin/org/checkerframework/checker/mungo/typestate/graph/exceptions/UnusedStates.kt
package org.checkerframework.checker.mungo.typestate.graph.exceptions

import org.checkerframework.checker.mungo.typestate.ast.TStateNode

class UnusedStates(private val unusedStates: List<TStateNode>) : TypestateError() {
  override val message: String
    get() = String.format("Unused states in %s: %s", unusedStates[0].pos.basename, unusedStates.map { it.name }.joinToString("; "))

  companion object {
    const val serialVersionUID = 0L
  }
}
